pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -315,22 +318,36 @@ func (_ OutboundProxyGenerator) determineRoutes(proxy *model.Proxy, outbound *ku
 		return clusters
 	}
 
-	for _, http := range route.Spec.GetConf().GetHttp() {
-		route := envoy_common.Route{
-			Match:    http.Match,
-			Modify:   http.Modify,
-			Clusters: clustersFromSplit(http.GetSplitWithDestination()),
+	addRoutes := func(route *mesh_core.TrafficRouteResource, rateLimit *mesh_proto.RateLimit, routes envoy_common.Routes) envoy_common.Routes {
+		for _, http := range route.Spec.GetConf().GetHttp() {
+			route := envoy_common.Route{
+				Match:     http.Match,
+				Modify:    http.Modify,
+				RateLimit: rateLimit,
+				Clusters:  clustersFromSplit(http.GetSplitWithDestination()),
+			}
+			routes = append(routes, route)
 		}
-		routes = append(routes, route)
+
+		if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
+			cfs := clustersFromSplit(defaultDestination)
+			route := envoy_common.Route{
+				Match:     nil,
+				Clusters:  cfs,
+				RateLimit: rateLimit,
+			}
+			routes = append(routes, route)
+		}
+		return routes
 	}
 
-	if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
-		cfs := clustersFromSplit(defaultDestination)
-		route := envoy_common.Route{
-			Match:    nil,
-			Clusters: cfs,
+	if len(proxy.Policies.RateLimits.Outbound[oface]) < 1 {
```

Comment:
```suggestion
if len(proxy.Policies.RateLimits.Outbound[oface]) == 0 {
```

Nit: this seems more conventional.